### PR TITLE
Update actor and messages

### PR DESCRIPTION
**Pull request by cmgrd**
Adds Actor concept and uses this in message generators.

#539 
#541 - Only partially fixed

### CODE REVIEW

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/ActorAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            var actors = _context.Actors
+                .Where(actor =>
+                    actor.IdentificationNumber == number &&
+                    actor.IdentificationType == type)
+                .ToList();
+
+            var actor = actors.FirstOrDefault(actor => actor.Roles.Contains(role));
+            if (actor == null) return null;
+
+            actor.SetRole(role);
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/ActorAccessor.cs:50]: What is the purpose of this line ? A getter function should not have side effects

File: source/Energinet.DataHub.MeteringPoints.Domain/Actors/Actor.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            if (role == null!) throw new ArgumentNullException(nameof(role));
+
+            if (!Roles.Contains(role))
+            {
+                throw new InvalidOperationException($"Actor doesn't have the role: {role.Name}");
+            }
+
+            CurrentRole = role;
+        }
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Domain/Actors/Actor.cs:66]: I would like to se some tests that describe the purpose and functionality of this logic

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/ActorAccessor.cs (hunk 1 above)
Comment (cmgrd) [reply source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/ActorAccessor.cs]: Removed

File: source/Energinet.DataHub.MeteringPoints.Domain/Actors/Actor.cs (hunk 2 above)
Comment (cmgrd) [reply source/Energinet.DataHub.MeteringPoints.Domain/Actors/Actor.cs]: Removed

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/Actors/ActorEntityConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                .HasConversion(
+                    toDbValue => string.Join(",", toDbValue
+                        .Select(role => role.Name)
+                        .ToList()),
+                    fromDbValue => new Collection<Role>(fromDbValue
+                        .Split(",", StringSplitOptions.None)
+                        .Select(EnumerationType.FromName<Role>)
+                        .ToList()));
+
+            builder.Ignore("CurrentRole");
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/DataAccess/Actors/ActorEntityConfiguration.cs:59]: According to DC this is obsolete?

**Review by Korgath — CHANGES_REQUESTED** [review]

**Review by Korgath — APPROVED** [review]

**Pull request merged**